### DISCUSSION TITLE
Showing configured model from desktop on mobile with the new changes

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#4275] [Q&A]
If a user configured a model on their desktop device, what would be the best way to show the model with the configured changes on a users mobile device and from there in AR? 

Up until now I usually kept track of the changes made by making variables and saving them in a URL link, then I would generate a QR code with that URL link and have the page read the values from the link and update the model manually, but that seems like a bad way to do it if there are a lot of changes... Any suggestions on how to do this better?

for example I would make it like this:

`

    // these vars would be updated with according numbers based on users choices

    var wood = 1;
    var seat = 1;

    function generateQRCode(value) {
      if(qrcode === undefined){
        qrcode = new QRCode( document.getElementById("qrcode"), {
          text: "http://link/?wood=" + wood + "&seat=" + seat,
          width: 256,
          height: 256,
          colorDark : "#000000",
          colorLight : "#ffffff",
          correctLevel : QRCode.CorrectLevel.H
        });
      }else{
        qrcode.clear();
        qrcode.makeCode("http://link/?wood=" + wood + "&seat=" + seat);
      }
    }

    // on page load get the updated variables (users choices)
    
    const queryString = window.location.search;
    const urlParams = new URLSearchParams(queryString);
    var woodUrl = urlParams.get('wood');
    var seatUrl = urlParams.get('seat');
    
    // on page load i would call this function to update the model
    
    function UpdateModel() {
        if(seatUrl == 1){
          Seat1();  // update seat material to choice 1
        }else if (seatUrl == 2){
          Seat2(); // update seat material to choice 2
        }
        
        if(woodUrl == 1){
          Wood1();  // update wood material to choice 1
        }else if (woodUrl == 2){
          Wood2();  // update wood material to choice 2
        }
      }
    
`

**Comment by WadeMorrison** [comment]
Hello danilokrulj!

I think the best way to do this is using a piping server, or your own server that you can exportScene() and then have the user scan a QR code that will open up another <model-viewer> page that will fetch that uploaded scene from the server. It's basically how MV's editor works when you click "View On Mobile"

I'm not great at programming, still relatively new to it, but after picking apart [karanganesan](https://github.com/karanganesan/model-viewer-bridge)'s example (doesn't seem to work anymore) and also [MV's own code](https://github.com/google/model-viewer/tree/master/packages/space-opera/src/components/mobile_view), I've came up with this [solution](https://github.com/google/model-viewer/discussions/4072#discussioncomment-5756372) that works for me at least! 

Basically the configurator waits for a "ping" from the mobile page which will ping it as soon as a user opens that page, from there it will then fire the exportScene() function and then upload it to a piping server, and then the mobile page will fetch the blob and set it as it's "src". I've tried it without the "ping" but it never worked for me, kept saying that the connection was already established for some reason. If you have any critiques/better ways to write this let me know as I'm trying to learn as well!

**Reply by elalish** [reply]
Agreed, that can certainly work. Honestly your URL params solution seems perfectly reasonable to me as well - if it's complicated you might try defining it in JSON, stringifying the result and using that as your single param.

**Reply by danilokrulj** [reply]
Thank you for the suggestions! I will try them and get back with the feedback. :) Much appreciated!
